src/parser/xasl_generation.c

```diff
@@ -18187,6 +18197,9 @@ pt_to_insert_xasl (PARSER_CONTEXT * parser, PT_NODE * statement)
 	  OID_SET_NULL (&xasl->creator_oid);
 	}
 
+      assert (locator_is_class (class_obj, DB_FETCH_QUERY_READ) > 0);
+      (void) au_fetch_class (class_obj, &smclass, AU_FETCH_READ, AU_SELECT);
```

Comment:
I think you can put the code below right away
```
	  if (smclass && smclass->stats)
	    {
	      if (smclass->tde_algorithm != TDE_ALGORITHM_NONE)
		{
		  xasl->includes_tde_class = 1;
		}
	    }
```

Reply:
Thank you, It is better. I've reflected it in https://github.com/CUBRID/cubrid/pull/2450/commits/902f6e1b54f2e65459e14641305b56a149d71a10.
With the change you suggested, I found out that it was wrong to check `smclass->stats`, so I removed it. Please check it.